doc/RELEASE.adoc

```diff
@@ -0,0 +1,120 @@
+
+We follow https://semver.org[semantic versioning] for the packages, where version numbers have a `major.minor.patch` pattern with the following semantics:
+
+- A major release (e.g., 3.5.2 -> 4.0.0) may contain arbitrary changes to the API (subject to the backwards compatibility requirements described below).
+- A minor release (e.g., 3.5.2 -> 3.6.0) may contain changes such as new features, which technically can be breaking, but are unlikely to cause breakage in practice, and if it does, the fix should be straightforward (e.g., by renaming).
++
+A minor release is also allowed to change the observable behaviors of functions that users should not rely on. For example, if a function returns a list of things and makes no promise on the order of elements in the returned list, then a minor release may change the order.
+- A patch release (e.g., 3.5.2 -> 3.5.3) can only contain such things as simple bug fixes, performance improvements, and documentation updates.
+
+=== Version Qualifiers
+
+Version qualifiers, also known as pre-release versions, usually refer to alpha, beta, and release candidate. Typically, alpha and beta are pre-feature complete, while release candidates are post-feature complete.
+
+We do not tag or publish alpha or beta verions for libraries. Libraries are published on Hackage, and Hackage does not support version qualifiers. It is also uncommon for a library to publish pre-release versions (unless it is new or a complete rewrite).
+However, we may annotate individual functions in a library as alpha or beta if we deem it appropriate.
+
+We do tag release candidates, e.g,. 3.5.2-rc1, because additional non-CI tests and checks may need to be carried out.
+These tests and checks either are too expensive to run in CI, or require manual steps (e.g., security audit), but are prerequisites of making a release.
+Tagging release candidates makes it easy to identify the correct commit to run these tests and checks against.
+
+For executables, we may publish alpha, beta or release candidates at our discretion.
+
+== Release Frequency
+
+We will begin by making a new major or minor release every 4 weeks, and evaluate and adjust the frequency in the future.
+Ad-hoc releases can be made upon request.
+
+We only make patch releases for version `x.y`, if a cardano-node release candidate depends on `x.y.z`, and an issue is found during the node release process that requires patching plutus.
+In all other cases, we always start a new major or minor release from master.
+
+== Release Process
+
+=== Major and Minor Releases
+
+Suppose we are releasing version `x.y.0`.
+
+1. Update version numbers in cabal files to `x.y.0` (if it is not already the case).
+- Version bounds for packages in the same repo should be set to `==x.y.*`.
+2. Make sure the change log is up to date.
```

Comment:
How serious are we about that? If serious, we need to make it a policy that every change that isn't trivial goes into the changelog. @asutherlandus @michaelpj also would like to hear your opinion.

Reply:
For a library as upstream as Plutus, I think it is critical. Otherwise everyone downstream will need to slack us and ask for the right version to use!